src/sql/workbench/contrib/notebook/test/browser/notebookActions.test.ts

```diff
@@ -3,18 +3,104 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import * as TypeMoq from 'typemoq';
 import * as assert from 'assert';
-
-import { AddCellAction, ClearAllOutputsAction, CollapseCellsAction, TrustedAction, RunAllCellsAction, NewNotebookAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
-import { CellType } from 'sql/workbench/services/notebook/common/contracts';
-import { INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import * as azdata from 'azdata';
+import * as sinon from 'sinon';
+import { TestConfigurationService } from 'sql/platform/connection/test/common/testConfigurationService';
+import { AddCellAction, ClearAllOutputsAction, CollapseCellsAction, KernelsDropdown, msgChanging, NewNotebookAction, noKernelName, RunAllCellsAction, TrustedAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
+import { ClientSessionStub as stubs_ClientSessionStub, ContextViewProviderStub, NotebookComponentStub, NotebookModelStub as stubs_NotebookModelStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { NotebookEditorStub } from 'sql/workbench/contrib/notebook/test/testCommon';
 import { ICellModel, INotebookModel } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { IStandardKernelWithProvider } from 'sql/workbench/services/notebook/browser/models/notebookUtils';
+import { INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import { CellType } from 'sql/workbench/services/notebook/common/contracts';
+import * as TypeMoq from 'typemoq';
+import { Emitter } from 'vs/base/common/event';
+import { TestCommandService } from 'vs/editor/test/browser/editorTestServices';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+import { IConfigurationChangeEvent, IConfigurationOverrides, IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { TestInstantiationService } from 'vs/platform/instantiation/test/common/instantiationServiceMock';
 import { INotificationService } from 'vs/platform/notification/common/notification';
 import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
-import { ICommandService } from 'vs/platform/commands/common/commands';
-import { TestCommandService } from 'vs/editor/test/browser/editorTestServices';
-import { NotebookComponentStub } from 'sql/workbench/contrib/notebook/test/stubs';
+import { workbenchInstantiationService } from 'vs/workbench/test/browser/workbenchTestServices';
+
+class ClientSessionStub extends stubs_ClientSessionStub {
```

Comment:
Yes, these overrides are specific to the usages in the tests in this class. I can use a new name but even that leads to confusion as to how are different test objects for similar things different. The pattern that made sense to me was to create a subclass that is needed specifically for a bunch of tests by the same name. So that we search for that name in our codebase we are able to find all similar usages and can do the right customizations needed when writing new tests. The idea is that the class in the stub.ts are essentially classes with methods that throw  'not implemented' exception. When for a bunch of tests you need a couple of methods to do something specific then you create a subclass and the best name I can think of was to reuse the same name because these new subclasses are still stubs for the most part with some customizations. However, if there is an agreement, I can call these subclasses with a prefix(or a suffix) of 'Test', so this class would be named as TestClientSession.

Reply:
Renaming seems good to me. If the class is implementing methods, then it's not really a stub anymore. Plus having duplicate classes floating around that do different things seems a bit confusing.

Reply:
Renamed